### PR TITLE
Create project idea

### PR DESCRIPTION
**Pull request by ayan-biswas0412**
## Describe your project in 1-3 sentences

I am building an intelligent door access control system which will have the following features 
  * Camera-based face recognition
  * RFID-based card lock system for user authentication.

## Hack Club Slack handle

[@AYAN BISWAS](https://hackclub.slack.com/team/U01584DM9PH)

## Checklist before submitting proposal:

- [x] I am a current high school student (or younger).
- [x] I have submitted an RSVP at <https://hackclub.com/winter>
- [x] I am a member of the Hack Club Slack (Join at <https://hackclub.com/slack>)
- [x] I have self-reviewed my project proposal.

### CODE REVIEW

File: ayan-biswas0412.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    * Grant or deny access to the door based on the result of the facial recognition
+5.Test the system to ensure it is functioning properly.
+6. Install the camera, servo motor, and lock mechanism near the desired door.
+
+
+## Budget
+
+| Product         | Supplier/Link                         | Cost   |
+| --------------- | ------------------------------------- | ------ |
+| Raspberry Pi 3 - Model B - ARMv8 with 1G RAM   | [https://www.adafruit.com/product/3055](https://www.adafruit.com/product/3055) | $35  |
```
Comment (bellesea) [inline ayan-biswas0412.md:37]: it looks like this is out of stock, do you have another source/site where you'd be able to purchase the pi?

Comment (bellesea) [reply ayan-biswas0412.md:37]: also, do you need an SD card?

File: ayan-biswas0412.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+1. Gather materials and tools needed:
+    * Raspberry Pi microcomputer
+    * Raspberry Pi camera module
+    * Servo motor
+    * Lock mechanism (such as a deadbolt or latch bolt)
+    * Wires and connectors
+    * Power supply
+2. Set up the Raspberry Pi and camera module according to the manufacturer's instructions.
+3. Connect the servo motor to the Raspberry Pi and wire it to control the lock mechanism.
+4. Write and upload code to the Raspberry Pi that will:
```
Comment (bellesea) [inline ayan-biswas0412.md:25]: how do you plan on running the facial recognition algorithm?

File: ayan-biswas0412.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+| Product         | Supplier/Link                         | Cost   |
+| --------------- | ------------------------------------- | ------ |
+| Raspberry Pi 3 - Model B - ARMv8 with 1G RAM   | [https://www.adafruit.com/product/3055](https://www.adafruit.com/product/3055) | $35  |
+| Raspberry Pi Camera Board v2 - 8 Megapixels   | [https://www.adafruit.com/product/3099](https://www.adafruit.com/product/3099) | $29.95  |
+| Micro servo   | [https://www.adafruit.com/product/169](https://www.adafruit.com/product/169) | $5.95  |
+| Rotating Door Lock DC Motor| [https://www.adafruit.com/product/3881](https://www.adafruit.com/product/3881) | $2.50  |
+| Wires and connectors  | [https://www.adafruit.com/product/1957](https://www.adafruit.com/product/1957) | $1.95  |
+| 5V 2.5A Switching Power Supply with 20AWG MicroUSB Cable   | [https://www.adafruit.com/product/1995](https://www.adafruit.com/product/1995) | $8.25  |
+| HDMI 5" 800x480 Display Backpack - With Resistive Touchscreen   | [https://www.adafruit.com/product/2260](https://www.adafruit.com/product/2260) | $74.95  |
+| Speaker   | [https://www.adafruit.com/product/3885](https://www.adafruit.com/product/3885) | $5.95  |
```
Comment (bellesea) [inline ayan-biswas0412.md:44]: what's the role of the speaker?

File: ayan-biswas0412.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+1. Gather materials and tools needed:
+    * Raspberry Pi microcomputer
+    * Raspberry Pi camera module
+    * Servo motor
+    * Lock mechanism (such as a deadbolt or latch bolt)
+    * Wires and connectors
+    * Power supply
+2. Set up the Raspberry Pi and camera module according to the manufacturer's instructions.
+3. Connect the servo motor to the Raspberry Pi and wire it to control the lock mechanism.
```
Comment (bellesea) [inline ayan-biswas0412.md:24]: do you plan on breadboarding this or soldering the components together? and, do you have the tools you need?

**Review by bellesea — CHANGES_REQUESTED** [review]
@ayan-biswas0412 this is really really cool! i'm wondering, are you following any sort of guide for this?

and also left some specific comments hehe :)

thank you!

File: ayan-biswas0412.md (hunk 1 above)
Comment (ayan-biswas0412) [reply ayan-biswas0412.md:37]: No I don't need an SD card, and I have Amazon as my alternative from where I will be purchasing if this goes out of stock in adafruit.

File: ayan-biswas0412.md (hunk 2 above)
Comment (ayan-biswas0412) [reply ayan-biswas0412.md:25]: I have my algorithm ready and it is part of my previous work which I will be using here.

File: ayan-biswas0412.md (hunk 3 above)
Comment (ayan-biswas0412) [reply ayan-biswas0412.md:44]: It is for making a sound like -> 
Access Granted/Denied for the user

File: ayan-biswas0412.md (hunk 4 above)
Comment (ayan-biswas0412) [reply ayan-biswas0412.md:24]: I plan to solder them together, and yes I have necessary soldering iron and other stuffs in my home.

**Comment by ayan-biswas0412** [comment]
> @ayan-biswas0412 this is really really cool! i'm wondering, are you following any sort of guide for this?
> 
> and also left some specific comments hehe :)
> 
> thank you!

Hi @bellesea thanks for reviewing my file, I have addressed all of your comments, can you please check the previous conversations?

And yes, I got motivation for the project from some youtube videos and online tutorials.

File: ayan-biswas0412.md (hunk 1 above)
Comment (bellesea) [reply ayan-biswas0412.md:37]: sounds good, thanks!

File: ayan-biswas0412.md (hunk 3 above)
Comment (bellesea) [reply ayan-biswas0412.md:44]: ahh cool!!

File: ayan-biswas0412.md (hunk 4 above)
Comment (bellesea) [reply ayan-biswas0412.md:24]: great, thank you!

**Review by bellesea — APPROVED** [review]
@ayan-biswas0412 this looks awesome! can't wait to see this being worked on hehe :)

an awesome member of @hackclub/bank-ops will be reaching out next week with next steps!

btw, could you please add to the file and credit some resources you're referencing? thank you :)

**Comment by ayan-biswas0412** [comment]
> @ayan-biswas0412 this looks awesome! can't wait to see this being worked on hehe :)
> 
> an awesome member of @hackclub/bank-ops will be reaching out next week with next steps!
> 
> btw, could you please add to the file and credit some resources you're referencing? thank you :)

Yeah I have attached some references in my file. Thanks for approving the PR @bellesea

**Review by MelSmith104 — COMMENTED** [review]
hey @ayan-biswas0412 this projects looks so cool! i love that you want to build it because of your interest in improving security - im really excited to see you make this! im dm'ing you on slack right now to chat about next steps!

**Comment by ayan-biswas0412** [comment]
Thanks @MelSmith104 for merging the PR.

**Pull request merged**